### ISSUE TITLE
Are files encrypted to the Tor exit node owner?

### ISSUE THREAD

**Issue by ThePwnd** [onionshare/onionshare#1652]
This might be due to a misunderstanding of the Tor network on my part.  It is my understanding that when browsing the web over the Tor network, the owner of the exit node your computer reaches can see the data that your computer is sending to the website, and the data that the website is sending back.  It just can't see who the origin of the request across the Tor network.  If that is the case, does OnionShare have the same vulnerability?  Can the owner of the exit node still see data in the file being transferred?

If I'm wrong, and the files are encrypted to the owner of the exit node, does that change if I turn off the private key?

**Comment by mig5** [comment]
Hi @ThePwnd

> It is my understanding that when browsing the web over the Tor network, the owner of the exit node your computer reaches can see the data that your computer is sending to the website, and the data that the website is sending back.

That is true when accessing regular websites e.g a site on a .com, .net domain etc - what we sometimes refer to as a 'clearnet' website. (Note that if the site is using HTTPS, the exit node still doesn't know what was contained in the request/response because it's encrypted with the HTTPS certificate)

>  If that is the case, does OnionShare have the same vulnerability? Can the owner of the exit node still see data in the file being transferred?

OnionShare is not affected by this because the address (ending in `.onion`) is an 'onion service' . When visiting .onion addresses, there *is* no 'exit node' at all. It remains encrypted end-to-end (yes, despite the 'http://' in the address bar) and the traffic never 'exits' the Tor network at all. The only service that can decrypt the request to the .onion service is the onion service itself.

> If I'm wrong, and the files are encrypted to the owner of the exit node, does that change if I turn off the private key?

The Private Key feature is not related to the encryption of the traffic in any way. Traffic back and forth with the .onion service is encrypted no matter what. The Private Key feature just adds the ability to enforce having to enter a password to visit the onion service. 

If someone discovers your long `.onion` address, the Private Key feature means they need to also enter a password to load the service. If you turn off the Private Key feature ('This is a public service'), then they don't need to enter a password (private key) and can visit the .onion address straight away.

In summary:
 * Traffic to onion services is always encrypted end to end, no matter what
 * There is no 'exit' node when using .onion services, so that concern doesn't apply here
 * Private Key is just about putting password protection over the service, which is good if you want your `.onion` address to remain secret and are worried about what happens if it gets exposed/leaked
 
I hope this helps! Let me know if you have any further questions!
